File: fixtures/drizzle/sqlalchemy/post.py
from sqlalchemy import DateTime, Enum, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship
from datetime import datetime

from .base import Base


class Post(Base):
    __tablename__ = "posts"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)

    title: Mapped[str] = mapped_column(String, nullable=False)

    slug: Mapped[str] = mapped_column(String, unique=True, nullable=False)

    content: Mapped[str] = mapped_column(String, nullable=False)

    status: Mapped[str] = mapped_column(Enum("DRAFT", "PUBLISHED", "ARCHIVED", name="post_status"), nullable=False, default="DRAFT")

    views: Mapped[int] = mapped_column(Integer, nullable=False, default=0)

    author_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False)

    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, server_default=func.now())

    updated_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, onupdate=func.now())

    author: Mapped["User"] = relationship("User", back_populates="posts")

    comments: Mapped[list["Comment"]] = relationship("Comment", back_populates="post")

    post_tags: Mapped[list["PostTag"]] = relationship("PostTag", back_populates="post")

File: examples/active-record/app/models/post.rb
class Post < ApplicationRecord
  self.inheritance_column = nil

  attribute :body, default: ""
  attribute :views, default: 0
  alias_attribute :created_at, :createdAt
  alias_attribute :updated_at, :updatedAt

  enum :status, { draft: "DRAFT", published: "PUBLISHED", archived: "ARCHIVED" }, default: :draft, validate: true
  enum :visibility, { public: "public", private: "private", unlisted: "UNLISTED" }, default: :public, prefix: true,
                                                                                    validate: true

  validates :title, presence: true, length: { maximum: 80 }
  validates :views, numericality: { greater_than_or_equal_to: 0 }

  belongs_to :author, class_name: "User", optional: true, inverse_of: :posts
  has_many :post_tags, dependent: :destroy
  has_many :comments, dependent: :destroy
  has_and_belongs_to_many :categories, join_table: "_CategoryToPost", foreign_key: "B", association_foreign_key: "A"

  has_many :tags, through: :post_tags
  scope :recent, -> { order(created_at: :desc) }
end
